contributors/design-proposals/testing/test-frameworks-repo.md

```diff
@@ -0,0 +1,65 @@
+# New git repo: k8s.io/testframeworks
+
+Status: Pending
+
+Version: N/A
+
+Implementation Owner: TBD
+
+## Motivation: Why do we need a new repo?
+
+To avoid an vendoring cycle.
+
+Some folks in Sig-CLI have been writing an [integration testing
+framework](https://github.com/kubernetes/kubectl/tree/master/pkg/framework/test)
+which they would like to use to simplify some of their integration tests which
+live in `k8s.io/kubernetes`. The framework currently lives in `k8s.io/kubectl`.
+Since `k8s.io/kubectl` already vendors components _from_ `k8s.io/kubernetes`, it 
+cannot be vendored _into_ `k8s.io/kubernetes`. 
+
+It therefore seems like a good idea to create a new repo to contain testing
+frameworks. The intent is that this repo could be vendored anywhere.
+
+## Proposal
+
+We propose a new git repository named `k8s.io/testframeworks`. This would be
+backed by a github repository `github.com/kubernetes/testframeworks`.
+
+### What should live in it?
```

Comment:
We might want to make explicit which other `k8s.io/` repos this would be allowed to import

Reply:
Good call. Which repos would folks like? :)

Reply:
Ideally, none.

Reply:
But the e2e might depend on the client? (Or does it shell out?)

Reply:
If it does, then I don't think it can live outside of k/k.